### PR TITLE
Make more exact dialyzer diagnostics by utilizing column number

### PR DESCRIPTION
**Pull request by plux**
### Description

Dialyzer diagnostics will now use the same range calculation as compiler diagnostics.

Follow up to suggestion in PR of #1006

### CODE REVIEW

**Review by gomoripeti — APPROVED** [review]
awesome 🎉

File: apps/els_lsp/test/els_server_SUITE.erl [hunk 1]
```diff
@@ -108,6 +108,7 @@ wait_until_no_lens_jobs() ->
 
 -spec get_current_lens_jobs() -> [pid()].
 get_current_lens_jobs() ->
-  #{internal_state := InternalState} = sys:get_state(els_code_lens_provider),
+  #{internal_state := InternalState} =
+    sys:get_state(els_code_lens_provider, 30 * 1000),
```
Comment (robertoaloi) [inline apps/els_lsp/test/els_server_SUITE.erl:112]: Do we know why this takes so long?

**Review by robertoaloi — APPROVED** [review]

**Comment by robertoaloi** [comment]
Thanks! We should probably figure out why the `sys:get_state/1,2` takes so long.

File: apps/els_lsp/test/els_server_SUITE.erl (hunk 1 above)
Comment (plux) [reply apps/els_lsp/test/els_server_SUITE.erl:112]: No we should look into it!

**Pull request merged**